rkt/images.go

```diff
@@ -224,49 +222,101 @@ func (f *fetcher) fetchSingleImage(img string, asc string, discover bool) (strin
 		return "", fmt.Errorf("unable to access %q: %v", img, err)
 	}
 
-	if discover && u.Scheme == "" {
+	switch u.Scheme {
+	// image name
+	case "":
+		return f.fetchImageByName(img, ascFile)
+	case "http", "https", "file", "docker":
+		return f.fetchImageByURL(u, ascFile)
+	}
+
+	return "", fmt.Errorf("rkt only supports image name and http, https, docker or file URLs (%s)", img)
+}
+
+func (f *fetcher) fetchImageByName(img string, ascFile *os.File) (string, error) {
+	// check the store
+	if !f.forceRemote {
+		key, err := getStoreKeyFromApp(f.s, img)
+		if err != nil {
+			switch err.(type) {
+			// ignore the error if it's a store.ACINotFoundError
+			case store.ACINotFoundError:
+			default:
+				return "", err
+			}
+		} else {
+			stderr("rkt: using image from local store for image name %s", img)
+			return key, nil
+		}
```

Comment:
maybe change to:
```go
if err == nil {
    return
}
switch err {
}
``` 
So that we can [avoid the else](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow)?